samples/crypto/spake2p/src/main.c

```diff
@@ -0,0 +1,287 @@
+#define APP_ERROR_MESSAGE   "Example exited with error!"
+
+/* w0 || w1 */
+static const uint8_t key_pair[] = {0x54, 0x8E, 0xC1, 0x42, 0xE2, 0x27, 0x90, 0x23, 0x7C, 0x67, 0xA8,
+				   0x88, 0x49, 0xE8, 0x61, 0xD3, 0x77, 0x00, 0x5F, 0x0A, 0x5C, 0x33,
+				   0x88, 0xF9, 0xAF, 0xA1, 0xC2, 0xFA, 0x58, 0xC7, 0xDA, 0x51, 0x35,
+				   0x99, 0xF8, 0x67, 0x1D, 0xBB, 0x67, 0x04, 0xA2, 0xC6, 0x3A, 0x78,
+				   0x4F, 0xC9, 0x5C, 0xD2, 0x8E, 0xBC, 0x55, 0x2E, 0xA4, 0x79, 0x98,
+				   0xB9, 0x18, 0x69, 0x9A, 0xB9, 0x3F, 0x4F, 0x7A, 0xD7};
+
+/* w0 || L */
+static const uint8_t public_key[] = {
+	0x54, 0x8E, 0xC1, 0x42, 0xE2, 0x27, 0x90, 0x23, 0x7C, 0x67, 0xA8, 0x88, 0x49, 0xE8,
+	0x61, 0xD3, 0x77, 0x00, 0x5F, 0x0A, 0x5C, 0x33, 0x88, 0xF9, 0xAF, 0xA1, 0xC2, 0xFA,
+	0x58, 0xC7, 0xDA, 0x51, 0x04, 0x81, 0x43, 0x3D, 0xC5, 0x93, 0xC9, 0x46, 0xC9, 0x37,
+	0xD9, 0x90, 0x26, 0xDD, 0x42, 0x14, 0x40, 0xE1, 0xC8, 0x7D, 0x0E, 0xC4, 0x94, 0x8B,
+	0xFF, 0x59, 0xEA, 0xF4, 0x77, 0xE3, 0x35, 0xE5, 0x52, 0x49, 0x66, 0xB2, 0x03, 0x31,
+	0x37, 0xD8, 0x4C, 0x65, 0x56, 0xDE, 0x07, 0x31, 0x57, 0x5C, 0xD2, 0x95, 0xC9, 0x75,
+	0x12, 0x4F, 0x52, 0x13, 0x25, 0xF7, 0x80, 0x01, 0xEC, 0xBE, 0x67, 0xE8, 0xB7};
+
+static psa_status_t send_message(psa_pake_operation_t *from, psa_pake_operation_t *to,
+				 psa_pake_step_t step)
+{
+	uint8_t data[1024];
+	size_t length;
+
+	psa_status_t status = psa_pake_output(from, step, data, sizeof(data), &length);
+
+	if (status) {
+		return status;
+	}
+	PRINT_HEX("send_message", data, length);
+	status = psa_pake_input(to, step, data, length);
+	return status;
+}
+
+static psa_status_t pake_setup(psa_pake_operation_t *op, psa_pake_role_t role, psa_key_id_t key,
+			       psa_pake_cipher_suite_t *cipher_suite)
+{
+	psa_status_t status;
+
+	status = psa_pake_setup(op, key, cipher_suite);
+	if (status != PSA_SUCCESS) {
+		LOG_INF("psa_pake_setup failed. (Error: %d)", status);
+		return status;
+	}
+
+	status = psa_pake_set_role(op, role);
+	if (status != PSA_SUCCESS) {
+		LOG_INF("psa_pake_set_role failed. (Error: %d)", status);
+		return status;
+	}
+
+	if (role == PSA_PAKE_ROLE_SERVER) {
+		status = psa_pake_set_user(op, "client", strlen("client"));
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_set_user failed. (Error: %d)", status);
+			return status;
+		}
+
+		status = psa_pake_set_peer(op, "server", strlen("server"));
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_set_peer failed. (Error: %d)", status);
+			return status;
+		}
+	} else {
+		status = psa_pake_set_user(op, "server", strlen("server"));
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_set_user failed. (Error: %d)", status);
+			return status;
+		}
+
+		status = psa_pake_set_peer(op, "client", strlen("client"));
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_set_peer failed. (Error: %d)", status);
+			return status;
+		}
+	}
+
+	return status;
+}
+
+int main(void)
+{
+	/* For matter-compatible spake2p this will be PSA_ALG_SPAKE2P_MATTER */
```

Comment:
Oh, no, I totally wouldn't have checked this if it wasn't for your comment Georgios, I'm blaming it on you LOL
```suggestion
	/* For Matter-compatible Spake2+, this will be PSA_ALG_SPAKE2P_MATTER */
```